File: Java/Strategy/Ducks/src/main/java/MallardDuck.java
package Strategy;

public class MallardDuck extends Duck {
	//Constructor
	public MallardDuck() {
		quackBehaviour =  new Quack();
		flyBehaviour = new FlyWithWings();
	}

	public void display() {
		System.out.println("I'm a real Mallard Duck");
	}

}

File: CSharp/Strategy.Duck/MallardDuck.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DesignPatterns.Strategy
{
    public class MallardDuck : Duck
    {
        public MallardDuck()
        {
            flyBehaviour = new FlyWithWings();
            quackBehaviour = new Quack();
        }

        public override void display()
        {
            Console.Out.WriteLine("I am a real Mallard Duck");
        }
    }
}
